tests/backup/backup_restore_basic_test.go

```diff
@@ -1442,3 +1442,220 @@ var _ = Describe("{BackupMultipleNsWithSameLabel}", func() {
+				err = CreateBackupLocation(provider, customBackupLocationName, backupLocationUID, credName, cloudCredUID, getGlobalBucketName(provider), orgID, "")
+				backupLocationMap[backupLocationUID] = customBackupLocationName
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating backup location %s", customBackupLocationName))
+			}
+		})
+
+		Step("Register source and destination cluster for backup", func() {
+			log.InfoD("Registering Source and Destination clusters and verifying the status")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			err = CreateSourceAndDestClusters(orgID, "", "", ctx)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+			clusterStatus, err = Inst().Backup.GetClusterStatus(orgID, SourceClusterName, ctx)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", SourceClusterName))
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", SourceClusterName))
+			clusterUid, err = Inst().Backup.GetClusterUID(ctx, orgID, SourceClusterName)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching [%s] cluster uid", SourceClusterName))
+		})
+
+		Step("Taking backup of applications", func() {
+			log.InfoD("Taking backup of applications")
+			var sem = make(chan struct{}, numberOfSimultaneousBackups)
+			var wg sync.WaitGroup
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			log.InfoD("Taking %d backups", numberOfBackups)
+			for backupLocationUID, backupLocationName := range backupLocationMap {
+				for _, namespace := range bkpNamespaces {
+					for i := 0; i < numberOfBackups; i++ {
+						sem <- struct{}{}
+						time.Sleep(timeBetweenConsecutiveBackups)
+						backupName := fmt.Sprintf("%s-%v", BackupNamePrefix, time.Now().Unix())
+						backupNames = append(backupNames, backupName)
+						wg.Add(1)
+						go func(backupName string) {
+							defer GinkgoRecover()
+							defer wg.Done()
+							defer func() { <-sem }()
+							err = CreateBackup(backupName, SourceClusterName, backupLocationName, backupLocationUID, []string{namespace},
+								labelSelectors, orgID, clusterUid, "", "", "", "", ctx)
+							dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying backup creation: %s", backupName))
+						}(backupName)
+					}
+					wg.Wait()
+				}
+			}
+			log.Infof("List of backups - %v", backupNames)
+		})
+
+		Step("Remove the backup locations where backups were taken", func() {
+			log.InfoD("Remove backup locations where backups were taken")
+			// Issue a remove backup location call
+			for backupLocationUID, customBackupLocationName = range backupLocationMap {
+				err := DeleteBackupLocation(customBackupLocationName, backupLocationUID, orgID, false)
+				dash.VerifySafely(err, nil, fmt.Sprintf("Deleting backup location %s", customBackupLocationName))
```

Comment:
Shouldn't this be `log,FailOnError()`, since if we fail to delete a backup location, there's no point in moving on with the test.

Reply:
@mkoppal-px Please share some info here ^

Reply:
@tthurlapati-px As I had mentioned earlier in slack as well, the main objective of the test case revolves around backup location deletion, so it is ok to use `dash.VerifyFatal` and not `log.FailOnError`

@kshithijiyer-px Please change it to `dash.VerifyFatal` because we wouldn't want to proceed with the test in case backup location deletion fails right?

Reply:
Yeah, I was basically alluding to either `dash.VerifyFatal` or `log.FailOnError`. I preferred `log.FailOnError` because this code is not the actual confirmation of deletion; this is just the command to delete. The actual verification of deletion happens later on in the code, _which is where_ `dash.VerifyFatal` is used.

Reply:
Done